### PR TITLE
docs: update usage guide for v3.0.0

### PR DESCRIPTION
**Pull request by daniel-sanche**
This PR contains two major docs changes in prep for v3.0.0:
- a major re-write of the usage guide, documenting everything I learned while working with the library over the last year
- an UPGRADING.md update to describe major/breaking changes

The upgrading guide can be rendered [here on GitHub](https://github.com/googleapis/python-logging/blob/v3_usage_guide/UPGRADING.md). To render the usage guide locally, run the following commands:

```
cd python-logging
nox -s docs
cd docs/_build_html
python3 -m http.server 8085
```

### CODE REVIEW

**Review by losalex — APPROVED** [review]
As we spoke earlier, this should be reviewed by writers as well

**Comment by arbrown** [comment]
>  cd docs/_build_html

This directory was `cd docs/_build/html` on my machine.  Not sure if it was a nox config issue or a typo.

File: docs/direct-lib-usage.rst [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+---------------
+
+Sinks allow exporting entries which match a given filter to Cloud Storage
+buckets, BigQuery datasets, or Cloud Pub/Sub topics.
+
+Cloud Storage Sink
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Make sure that the storage bucket you want to export logs too has
+``cloud-logs@google.com`` as the owner. See
```
Comment (arbrown) [inline docs/direct-lib-usage.rst:232]: Consider 'as an owner' or 'has owner permissions'

File: docs/direct-lib-usage.rst [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+Using Log Sinks
+---------------
+
+Sinks allow exporting entries which match a given filter to Cloud Storage
+buckets, BigQuery datasets, or Cloud Pub/Sub topics.
+
+Cloud Storage Sink
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Make sure that the storage bucket you want to export logs too has
```
Comment (arbrown) [inline docs/direct-lib-usage.rst:231]:
```suggestion
Make sure that the storage bucket you want to export logs to has
```

File: docs/direct-lib-usage.rst [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+``cloud-logs@google.com`` as the owner. See
+`Setting permissions for Cloud Storage`_.
+
+.. _Setting permissions for Cloud Storage: https://cloud.google.com/logging/docs/export/configure_export_v2#errors_exporting_to_cloud_storage
+
+Add ``cloud-logs@google.com`` as the owner of the bucket:
```
Comment (arbrown) [inline docs/direct-lib-usage.rst:237]: same as above

**Review by arbrown — APPROVED** [review]
Minor nits on pre-existing docs, otherwise looks good!

**Pull request merged**